Add size and fixedWidth attribute to fa-layers #79

paustint opened the pull request:
Added size and fixedWidth attribute to FaLayersComponent
moved the {host: class: 'fa-layers'} to a @HostBinding decorator to allow dynamic updates
Added new class list helper `faLayerClassList()` to build class names

Added unit tests
Added examples
Updated gitignore because it was not properly omitting .vscode directory

resolves #79

devoto13 commented on src/lib/layers/layers.component.ts:
```diff
@@ -1,15 +1,38 @@
+import { Component, Input, SimpleChanges, OnChanges, OnInit, HostBinding } from '@angular/core';
+import { SizeProp } from '@fortawesome/fontawesome-svg-core';
+import { faLayerClassList } from '../shared/utils/classlist.util';
+import { FaProps } from '../public_api';
```
Please change import to use `../shared/models/props.model`. Public API guard is not intended to be imported from.

paustint commented on src/app/example.component.html:
```diff
@@ -48,6 +48,17 @@ <h3>Layers</h3>
   </fa-layers>
 </div>
+<p>Increase the size of the layers container.</p>
+<div class="fa-4x">
```
@devoto13 - thanks for noticing this, it is a left over class which I have corrected.  I also corrected all the other examples that are using `fa-layers` to use the new attribute instead of setting `[ngClass]="['fa-fw']"` manually so that all the examples are consistent.

devoto13 replied:
Great, thanks for updating it! 
Can you also replace `class="fa-4x"` with `size="4x"` on `fa-layers` element, so it uses the newly added property?

paustint replied:
@devoto13 - I have implemented the requested changes.  Also, I removed the section I had initially added to the examples since the use-case and use of both `[fixedWidth]` and `size` was already demonstrated in the existing examples after making this change.

Also, our implementation on this (now and the prior implementation) does not allow custom CSS classes to be defined by the user (e.x. a user cannot add `class="foo"` or `[ng-class]="bar"` directly to the `fa-layers` component because we are overriding the CSS in our component.  Do we want to call this out anywhere in the documentation that any additional classes must be on some wrapper element?

paustint commented:
@devoto13 - Thank you so much for the review. I have corrected the import path, which was initially set using auto-import feature in VSCode and I did not realize there were multiple versions of the interface so I did not scrutinize/pay attention to the path.

I also made some other minor updates to the other uses of `<fa-layers>` in the examples.  Take a look and let me know if it it looks good.

Thanks again for your review.

paustint commented:
@devoto13 - I have implemented the requested changes. Also, I removed the section I had initially added to the examples since the use-case and use of both [fixedWidth] and size was already demonstrated in the existing examples after making this change.

Also, our implementation on this (now and the prior implementation) does not allow custom CSS classes to be defined by the user (e.x. a user cannot add class="foo" or [ng-class]="bar" directly to the fa-layers component because we are overriding the CSS in our component. Do we want to call this out anywhere in the documentation that any additional classes must be on some wrapper element?

devoto13 commented:
@paustint Thanks a lot for updates! Great work!

IMO the host class binding overriding classes set from outside of component is known Angular issue and we don't need to specifically mention it in the documentation.

@mlwilkerson I think this is good to go now. I'm going to merge it unless you have any other comments?

mlwilkerson commented:
@devoto13 No comments, but I haven't been watching this one. Roll with it when you think it's ready.

devoto13 commented:
@paustint Thanks for the contribution!
